File: python-wrapper/src/python/model/embed/AuthenticationClass.py
from typing import Union, Dict
from enum import Enum

class AuthenticationClass(Enum):
	DigitalId = "DIGITAL_ID"
	TwoFactorAuthentication = "TWO_FACTOR_AUTHENTICATION"
	ShortLivedToken = "SHORT_LIVED_TOKEN"
	ExternalAuthentication = "EXTERNAL_AUTHENTICATION"
	Password = "PASSWORD"
	LongLivedToken = "LONG_LIVED_TOKEN"

	def __serialize__(self) -> object:
		return self.value

	@classmethod
	def _deserialize(cls, data: Union[str, Dict[str, object]]) -> 'AuthenticationClass':
		if data == "DIGITAL_ID":
			return AuthenticationClass.DigitalId
		elif data == "TWO_FACTOR_AUTHENTICATION":
			return AuthenticationClass.TwoFactorAuthentication
		elif data == "SHORT_LIVED_TOKEN":
			return AuthenticationClass.ShortLivedToken
		elif data == "EXTERNAL_AUTHENTICATION":
			return AuthenticationClass.ExternalAuthentication
		elif data == "PASSWORD":
			return AuthenticationClass.Password
		elif data == "LONG_LIVED_TOKEN":
			return AuthenticationClass.LongLivedToken
		else:
			raise Exception(f"{data} is not a valid value for AuthenticationClass enum.")

__all__ = ['AuthenticationClass']

File: cardinal-sdk/src/commonMain/kotlin/com/icure/cardinal/sdk/model/embed/AuthenticationClass.kt
package com.icure.cardinal.sdk.model.embed

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlin.String

@Serializable
public enum class AuthenticationClass(
	internal val dtoSerialName: String,
) {
	@SerialName("DIGITAL_ID")
	DigitalId("DIGITAL_ID"),

	@SerialName("TWO_FACTOR_AUTHENTICATION")
	TwoFactorAuthentication("TWO_FACTOR_AUTHENTICATION"),

	@SerialName("SHORT_LIVED_TOKEN")
	ShortLivedToken("SHORT_LIVED_TOKEN"),

	@SerialName("EXTERNAL_AUTHENTICATION")
	ExternalAuthentication("EXTERNAL_AUTHENTICATION"),

	@SerialName("PASSWORD")
	Password("PASSWORD"),

	@SerialName("LONG_LIVED_TOKEN")
	LongLivedToken("LONG_LIVED_TOKEN"),
}
